### PR TITLE
added trigger jenkins; closes #195

### LINKED ISSUE #195: [server] please trigger jenkins

### PR DESCRIPTION
**Pull request by thuey**

### CODE REVIEW

File: internal/server/config/TriggerJenkins.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	"net/url"
+	"path"
+	"strconv"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/pleaseops/pleaseops/internal/execution"
+	"github.com/pleaseops/pleaseops/internal/server/key"
+)
+
+type TriggerJenkinsCluster struct {
```
Comment (jonnymbgood) [inline internal/server/config/TriggerJenkins.go]: Cluster? Maybe Server (TriggerJenkinsServer)?

File: internal/server/config/TriggerJenkins.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	Server         string `mapstructure:"server"`
```
Comment (jonnymbgood) [inline internal/server/config/TriggerJenkins.go]: Maybe URL instead of Server?

**Comment by thuey** [comment]
I was trying to match what we did for K8s. Do you think we should change
K8s to match?

**Comment by jonnymbgood** [comment]
With K8s we are connecting to a cluster. With jenkins a server. Just thinking we may want the type to be JenkinsServer instead of JenkinsCluster.

**Comment by thuey** [comment]
Technically, Jenkins could be a cluster, too. But I was actually mostly
thinking about how CheckKubernetesCluster is configured with "Server"
instead of "URL"

**Comment by jonnymbgood** [comment]
Ahh. I used Server in that case to match the nomenclature in kubeconfig. What do you think about using JenkinsServer as the type and URL in the place of server?

**Comment by thuey** [comment]
I'm good with that. What do you think about KubernetesCluster instead of
CheckKubernetesCluster?

**Comment by jonnymbgood** [comment]
If we can leave the action prefixes for now (i.e. CheckKubernetesCluster), we can cleanly transfer to the official types in #232 by creating resources (i.e. KubernetesCluster). Can we stick with TriggerJenkinsServer for this PR and change everything in one go in the other PR?

**Comment by jonnymbgood** [comment]
I agree with the name change btw.

**Comment by thuey** [comment]
Sounds good

**Pull request merged**